### PR TITLE
[Option] Support MultiArg options

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Teach SwiftOption to handle MultiArg options, including generating Options from Options.td, etc. Even the only MultiArg option in swift is a NoDriver option, still teach the OptionTable to parse such options to future proof the implementation.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
needed to support: https://github.com/apple/swift/pull/73601

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

File: Sources/SwiftOptions/OptionParsing.swift [hunk 1]
```diff
@@ -171,6 +171,17 @@ extension OptionTable {
...
         index += 1
+
+      case .multiArg:
+        if argument != option.spelling {
+          throw OptionParseError.unknownOption(
+            index: index - 1, argument: argument)
+        }
+        let endIdx = index + Int(option.numArgs)
+        parsedOptions.addOption(option, argument: .multiple(Array()))
+        arguments[index..<endIdx].map { String($0) }.forEach { parsedOptions.addInput($0) }
```
Comment (owenv) [inline Sources/SwiftOptions/OptionParsing.swift:182]: Will this index off the end of the array if the user doesn't provide enough args?

Comment (cachemeifyoucan) [reply Sources/SwiftOptions/OptionParsing.swift:186]: Good catch. Added a testcase for that too.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

**Review by owenv — APPROVED** [review]

**Pull request merged**